pootle/apps/pootle_store/models.py

```diff
@@ -1704,105 +1850,26 @@ def insert(i1, j1, j2):
             update_dbids = set([x['dbid'] for x in update_unitids.values()])
             common_dbids.update(update_dbids)
 
+            # Step N-1: mark obsolete units as such
+
             obsolete_dbids = [old_unitids[uid]['dbid']
                               for uid in old_unitid_set -
                                          old_obsolete_unitid_set -
                                          new_unitid_set]
-            for unit in self.findid_bulk(obsolete_dbids):
-                # Use the same (parent) object since units will
-                # accumulate the list of cache attributes to clear
-                # in the parent Store object
-                unit.store = self
-                if not unit.isobsolete():
-                    unit.makeobsolete()
-                    unit._from_update_stores = True
-                    unit.save()
-                    changes['obsolete'] += 1
+            obsoleted = self.mark_units_obsolete(obsolete_dbids)
 
-            if not overwrite:
-                # Get units that were modified after last sync
-                filter_by = {'store': self}
-                if self.last_sync_revision is not None:
-                    filter_by.update({'revision__gt': self.last_sync_revision})
-
-                # If a dbunit is obsolete
-                # then the dbunit should be resurrected in any case
-                modified_units = set(
-                    Unit.objects.filter(**filter_by).exclude(state=OBSOLETE)
-                                .values_list('id', flat=True).distinct()
-                )
+            changes['obsolete'] = obsoleted
```

Comment:
Is there some reason why this is not like the following?

```python
            changes['obsolete'] = self.mark_units_obsolete(obsolete_dbids)
```

It would avoid having an unnecessary variable and reduce another line of code.